app/Http/Controllers/Service/Admin/TradersController.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace App\Http\Controllers\Service\Admin;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\AdminNewUpdateTraderRequest;
+use App\Market;
+use App\Sponsor;
+use App\Trader;
+use App\User;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\View\View;
+use Throwable;
+
+class TradersController extends Controller
+{
+    /**
+     * List Traders
+     *
+     * @return Application|Factory|View
+     */
+    public function index()
+    {
+        $traders = Trader::with(['users', 'market'])->get();
+        // TODO : efficiency
+        $traders = $traders->sortBy(function ($trader) {
+            return $trader->market->sponsor->name . '#' .
+                $trader->market->name . '#' .
+                $trader->name;
+        });
+        return view('service.traders.index', compact('traders'));
+    }
+
+    /**
+     * Store a new Trader
+     *
+     * @param AdminNewUpdateTraderRequest $request
+     * @return RedirectResponse
+     * @throws Throwable
+     */
+    public function store(AdminNewUpdateTraderRequest $request)
+    {
+        try {
+            $trader = DB::transaction(function () use ($request) {
+                // find our market or die
+                $m = Market::findOrFail($request->input('market'));
+
+                // make a trader
+                $t = Trader::create([
+                    'name' => $request->name,
+                    'market_id' => $m->id,
+                ]);
+
+                // get our updated or new users as ids
+                $userIds = $this->createOrUpdateUsersFromInput(
+                    (array)$request->get('users')
+                );
+
+                // users should be an array; sync them to our trader
+                $t->users()->sync($userIds);
+                return $t;
+            });
+        } catch (Throwable $e) {
+            // Oops! Log that
+            Log::error('Bad transaction for ' . __CLASS__ . '@' . __METHOD__ . ' by service user ' . Auth::id());
+            Log::error($e->getMessage());
+            Log::error($e->getTraceAsString());
+            // Throw it back to the user
+            return redirect()->route('admin.traders.create')->withErrors('Creation failed - DB Error.');
```

Comment:
We'd probably want a more clear error message later. maybe spell DB so that we don't confuse Anna or Faith.

Reply:
refactoring that try/catch block is a good task for later; we appear to be doing a _lot_ of it in various places.